### ISSUE TITLE
How to add a SNI on server , please .

### ISSUE THREAD

**Issue by ingnelson** [paniy/Xray_bash_onekey#10]
Hello . 
Could you tell me how to add on the server configuración a SNI ( server name indication ) , please.

Because my ISP provider only let me connect to internet using a SNI .

Or could you add a feature that tell us if we want to add a SNI , please . It is the only way I can connect to internet.

The client configuration similar like this : 

{
  "dns": {
    "hosts": {
      "domain:googleapis.cn": "googleapis.com"
    },
    "servers": [
      "https://1.0.0.1/dns-query",
      {
        "address": "https+local://223.5.5.5/dns-query",
        "domains": [
          "full:ca1.vless.co"
        ],
        "skipFallback": true
      }
    ]
  },
  "inbounds": [
    {
      "listen": "127.0.0.1",
      "port": 2081,
      "protocol": "socks",
      "settings": {
        "auth": "noauth",
        "udp": true,
        "userLevel": 8
      },
      "sniffing": {
        "destOverride": [
          "http",
          "tls"
        ],
        "enabled": true,
        "metadataOnly": false
      },
      "tag": "socks"
    },
    {
      "listen": "127.0.0.1",
      "port": 9081,
      "protocol": "http",
      "settings": {
        "allowTransparent": true,
        "userLevel": 8
      },
      "sniffing": {
        "destOverride": [
          "http",
          "tls"
        ],
        "enabled": true,
        "metadataOnly": false
      },
      "tag": "http"
    },
    {
      "listen": "127.0.0.1",
      "port": 6451,
      "protocol": "dokodemo-door",
      "settings": {
        "address": "1.0.0.1",
        "network": "tcp,udp",
        "port": 53
      },
      "tag": "dns-in"
    }
  ],
  "log": {
    "loglevel": "error"
  },
  "outbounds": [
    {
      "protocol": "vless",
      "settings": {
        "vnext": [
          {
            "address": "ca1.vless.co",
            "port": 443,
            "users": [
              {
                "encryption": "none",
                "id": "81f1f510-3ca7-4734-8956-0f4fce670af5",
                "level": 8
              }
            ]
          }
        ]
      },
      "streamSettings": {
        "network": "ws",
        "security": "tls",
        "tlsSettings": {
          "serverName": "my.mintmobile.com"
        },
        "wsSettings": {
          "headers": {
            "Host": "my.mintmobile.com"
          },
          "path": "/fastssh?u\u003dmamamio\u0026p\u003d6147b8242bcbc\u0026"
        }
      },
      "tag": "proxy-global-2"
    },
    {
      "protocol": "freedom",
      "settings": {
        "domainStrategy": "UseIP"
      },
      "tag": "direct"
    },
    {
      "protocol": "freedom",
      "settings": {
        "domainStrategy": "UseIP"
      },
      "tag": "bypass"
    },
    {
      "protocol": "blackhole",
      "tag": "block"
    },
    {
      "protocol": "dns",
      "settings": {},
      "tag": "dns-out"
    }
  ],
  "policy": {
    "levels": {
      "8": {
        "connIdle": 300,
        "downlinkOnly": 1,
        "handshake": 4,
        "uplinkOnly": 1
      }
    },
    "system": {
      "statsOutboundDownlink": true,
      "statsOutboundUplink": true
    }
  },
  "routing": {
    "domainMatcher": "mph",
    "domainStrategy": "IPIfNonMatch",
    "rules": [
      {
        "inboundTag": [
          "dns-in"
        ],
        "outboundTag": "dns-out",
        "type": "field"
      }
    ]
  },
  "stats": {}
}

**Comment by paniy** [comment]
You can configure Nginx like this:
```
server {
        listen 443 ssl http2;
        listen [::]:443 ssl http2;
        **server_name           a.com;**


        **location /abca**
        {
              **proxy_pass http://127.0.0.1:1001;**
        }
}

server {
        listen 443 ssl http2;
        listen [::]:443 ssl http2;
        **server_name           b.com;**


        **location /abcb**
        {
              **proxy_pass http://127.0.0.1:1002;**
        }
}
```
and configure Xray like this:
   ```
"inbounds": [{
            **"port": 1001,**
            **"listen": "127.0.0.1",**
            "tag": "VLESS-ws-in",
            "protocol": "VLESS",
            "settings": {
                "clients": [{
                    "id": "UUID",
                    "level": 0,
                    "email": "ws@idleleo.com"
                }],
                "decryption": "none"
            },
            "streamSettings": {
                "network": "ws",
                "security": "none",
                "wsSettings": {
                    **"path": "/abca"**
                }
            }
        },
{
            **"port": 1002,**
            **"listen": "127.0.0.1",**
            "tag": "VLESS-ws-in",
            "protocol": "VLESS",
            "settings": {
                "clients": [{
                    "id": "UUID",
                    "level": 0,
                    "email": "ws@idleleo.com"
                }],
                "decryption": "none"
            },
            "streamSettings": {
                "network": "ws",
                "security": "none",
                "wsSettings": {
                    **"path": "/abcb"**
                }
            }
        }]
```

**Comment by ingnelson** [comment]
@paniy can you tell me how to create the server using SNI ? Using vless + ws + grpc . ?

I use a website to create a account like I tell you , but I want to install it on my VPS . 

WEBSITE :
https://www.fastssh.com/page/create-vless-account/server/704/vless-canada-1/


Like this account :

vless://81f1f510-3ca7-4734-8956-0f4fce670af5@ca1.vless.co:443/?type=ws&encryption=none&host=my.mintmobile.com&path=%2Ffastssh%3Fu%3Dtestmio%26p%3D614b60e899429%26&security=tls&sni=my.mintmobile.com#ca1.vless.co


Please .

**Comment by ingnelson** [comment]
@paniy could you add this feature on the script ? 
Because in America , Singapore , India , Africa and many other countries we use SNI ( usually the ISP website ) in the request  to get to internet . This way the ISP think we are visiting their website , but we are visiting other website.

**Comment by paniy** [comment]
The script itself supports SNI and may not be well compatible on some servers. Recently, I will add the function of manually entering the server IP to solve the problem you raised.

**Comment by ingnelson** [comment]
@paniy yes. And the script ask for the SNI when it is installing . Remember the SNI can be any website the user what . The SNI and the hostname do Not need to have  the same IP from the VPS .

Example  :
www.google.com 
www.chinamobileltd.com
www.chinatelecomglobal.com

I hope you add the feature soon . Many people will use around the world .

**Comment by paniy** [comment]
https://github.com/paniy/Xray_bash_onekey/commit/d31cf9eab817d16feb79997762370655d7324598 Please note that CNAME your domain name to the domain name provided by the server provider

**Comment by ingnelson** [comment]
@paniy i modified the DNS record on cloudflare , it sort of works . 
I just created a subdomain with a dns redoes  like this :
ISP domain + VPS domain .
Example : www.chinamobileltd.com.vpsdomain
.tk


But the script does not allow to use just a FAKE SNI  ( any website ) that does  not hhave the  VPS IP . 

I found a script  that  cann do it . 


https://github.com/wulabing/V2Ray_ws-tls_bash_onekey


VLESS+websocket+TLS+Nginx+WWebsit


If you check what the sspecial tabout the  "website "
